### PR TITLE
kernel/proc: Move copying of resources to `process_vforkThread`

### PR DESCRIPTION
**Pull request by astalke**
## Description

Move copying of resources from `process_copy` to `process_vforkThread`.

Fixes: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1077
JIRA: RTOS-839

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia32-generic-qemu

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: proc/process.c [hunk 1]
```diff
@@ -1377,6 +1377,10 @@ static void process_vforkThread(void *arg)
 		hal_cpuTlsSet(&current->tls, current->context);
 	}
 
+	if (proc_resourcesCopy(parent->process) < 0) {
```
Comment (agkaminski) [inline proc/process.c:1380]: Move this before `hal_cpuDisableInterrupts()` - there's no need to keep interrupts disabled during copy, but it makes system less responsive. Access to the parent resources is synchronized by `proc_resourcesCopy` itself.

Comment (agkaminski) [reply proc/process.c:1380]: Also, it might cause massive deadlock, as we might block on a taken mutex with interrupts disabled

**Review by agkaminski — DISMISSED** [review]
LGTM

File: proc/process.c [hunk 2]
```diff
@@ -1369,6 +1369,10 @@ static void process_vforkThread(void *arg)
 	hal_memcpy(&current->process->tls, &parent->process->tls, sizeof(hal_tls_t));
 	hal_memcpy(&current->tls, &parent->tls, sizeof(hal_tls_t));
 
+	if (proc_resourcesCopy(parent->process) < 0) {
+		proc_threadEnd();
```
Comment (agkaminski) [inline proc/process.c:1373]: Ah, sorry, missed that. We need to wakeup the parent here (like in `if` in 1355)

Comment (agkaminski) [reply proc/process.c:1373]: And free parentkstack

File: proc/process.c [hunk 3]
```diff
@@ -1369,6 +1369,17 @@ static void process_vforkThread(void *arg)
 	hal_memcpy(&current->process->tls, &parent->process->tls, sizeof(hal_tls_t));
 	hal_memcpy(&current->tls, &parent->tls, sizeof(hal_tls_t));
 
+	if (proc_resourcesCopy(parent->process) < 0) {
+		vm_kfree(current->parentkstack);
+
+		hal_spinlockSet(&spawn->sl, &sc);
+		spawn->state = -ENOMEM;
```
Comment (agkaminski) [inline proc/process.c:1376]: Theoretically proc_resourcesCopy might also return EINVAL

**Review by agkaminski — APPROVED** [review]

**Review by Darchiv — APPROVED** [review]

**Pull request merged**